test/integration/triggered_publisher.cc

```diff
@@ -637,3 +637,278 @@ TEST_F(TriggeredPublisherTest,
 
   EXPECT_EQ(0u, recvCount);
 }
+
+/////////////////////////////////////////////////
+/// Test for invalid service name. A service, `/srv-dummy-test` is advertised
+/// and the callback is also provided in this test. Everytime an input msg is
+/// published to `/in_14` topic, triggered_publisher plugin will call the
+/// service `srv-test`, specified in the test/worlds/triggered_publisher.sdf.
+/// However, since the two service names do not match, the callback provided in
+/// this test will not be invoked. Therefore, the pubCount, which is set to 10,
+/// will not equal to recvCount. The recvCount will be 0, since the callback
+/// isn't invoked.
+TEST_F(TriggeredPublisherTest,
+       IGN_UTILS_TEST_DISABLED_ON_WIN32(InvalidServiceName))
+{
+  transport::Node node;
+  auto inputPub = node.Advertise<msgs::Empty>("/in_14");
+  std::atomic<std::size_t> recvCount{0};
+
+  auto srvEchoCb = std::function<bool(const msgs::StringMsg &,
+      msgs::StringMsg &)>(
+      [&recvCount](const auto &req, auto &rep)
+      {
+        EXPECT_EQ(req.data(), "test");
+        if (req.data() == "test")
+        {
+          ++recvCount;
+          rep.set_data(req.data());
+          return true;
+        }
+        return false;
+      });
+
+  // Advertise a dummy service
+  std::string service = "/srv-dummy-test";
+  node.Advertise(service, srvEchoCb);
+
+  const std::size_t pubCount{10};
+  for (std::size_t i = 0; i < pubCount; ++i)
+  {
+    EXPECT_TRUE(inputPub.Publish(msgs::Empty()));
+    IGN_SLEEP_MS(100);
+  }
+
+  waitUntil(5000, [&]{return recvCount == 0u;});
+  EXPECT_EQ(recvCount, 0u);
+}
+
+/////////////////////////////////////////////////
+/// Test for triggering a service call in response to an input msg. A service,
+/// `srv-test` is advertised and the callback is also provided in this test.
+/// Everytime an input msg is published to `/in_14` topic, triggered_publisher
+/// plugin will call the service `/srv-test`, specified in the test/worlds/
+/// triggered_publisher.sdf. This time, the name of the services match. By
+/// publishing input msg 10 times, the service callback will also be invoked 10
+/// times. The `pubCount` is set to 10 and recvCount is increased everytime
+/// request data matches the string "test" inside the service callback. For a
+/// successful test, the pubCount will equal to recvCount.
+TEST_F(TriggeredPublisherTest,
+       IGN_UTILS_TEST_DISABLED_ON_WIN32(InputMessagesTriggerServices))
+{
+  transport::Node node;
+  auto inputPub = node.Advertise<msgs::Empty>("/in_14");
+  std::atomic<std::size_t> recvCount{0};
+
+  auto srvEchoCb = std::function<bool(const msgs::StringMsg &,
+      msgs::StringMsg &)>(
+      [&recvCount](const auto &req, auto &rep)
+        {
+          EXPECT_EQ(req.data(), "test");
+          if (req.data() == "test")
+            {
+              ++recvCount;
+              rep.set_data(req.data());
+              return true;
+            }
+          return false;
```

Comment:
Are these return values being checked anywhere? Are they needed?
If they're returning a value for a reason, they should be checked in the test. If they aren't there for any specific reason, then they aren't needed.